src/main/java/net/minecraftforge/world/WorldCapabilityData.java

```diff
@@ -0,0 +1,63 @@
+package net.minecraftforge.world;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.world.WorldProvider;
+import net.minecraft.world.WorldSavedData;
+import net.minecraftforge.common.capabilities.CapabilityDispatcher;
+import net.minecraftforge.common.util.INBTSerializable;
+
+public class WorldCapabilityData extends WorldSavedData
+{
+    private INBTSerializable<NBTTagCompound> serializable;
+    private NBTTagCompound data = null;
+    private int dimension;
+
+    public WorldCapabilityData(String name)
+    {
+        super(name);
+    }
+
+    public WorldCapabilityData(WorldProvider provider, INBTSerializable<NBTTagCompound> serializable)
+    {
+        super(fileNameForProvider(provider));
+        this.serializable = serializable;
+        this.dimension = provider.getDimension();
+    }
+
+	@Override
+    public void readFromNBT(NBTTagCompound nbt)
+    {
+	    if(nbt.hasKey("ForgeCaps"))
+	    {
+	        this.data = nbt.getCompoundTag("ForgeCaps");
+	        if(serializable != null)
+	            serializable.deserializeNBT(this.data);
+	    }
+	}
+
+    @Override
+    public void writeToNBT(NBTTagCompound nbt)
+    {
+        nbt.setInteger("dimension", this.dimension);
```

Comment:
I'm not very versed in WorldSavedData, but is this required? I thought WorldSavedData is per-world